### PR TITLE
(PUP-4190) Don't display credentials in output

### PR DESCRIPTION
**Pull request by hunner**
The url contains the credentials for a device. Don't print them in plaintext.

### CODE REVIEW

File: lib/puppet/application/device.rb [hunk 1]
```diff
@@ -169,7 +169,7 @@ def main
     end
     devices.each_value do |device|
       begin
-        Puppet.info "starting applying configuration to #{device.name} at #{device.url}"
+        Puppet.info "starting applying configuration to #{device.name}"
```
Comment (joshcooper) [inline lib/puppet/application/device.rb]: If you care about losing the debug context and `device.uri` is a ruby URI, you could use `#scheme`, `#host`, etc to extract out the public bits. You could also create a new URI, and maybe clear the user part with `#userinfo=`?

**Comment by branan** [comment]
Looks sane to me :+1:

**Comment by branan** [comment]
Actually, wait. Is it possible for device URLs to have path components, not just protocol/host/port? If so we might want to include those in the output too.

**Comment by hunner** [comment]
Updated for path.

**Pull request merged**